src/App.tsx

```diff
@@ -16,7 +63,27 @@ export const App: React.FC = () => {
       <div className="todoapp__content">
         <header className="todoapp__header">
           {/* this buttons is active only if there are some active todos */}
-          <button type="button" className="todoapp__toggle-all active" />
+          <button
+            type="button"
+            className="todoapp__toggle-all active"
+            onClick={() => {
+              setTodos(someTodos => {
+                const newTodos = [...someTodos]
+                  .map(todo => {
+                    const isSomeComplited = [...someTodos]
+                      .filter(someTodo => someTodo.completed).length !== someTodos.length;
+
+                    if (isSomeComplited) {
+                      return { ...todo, completed: true };
+                    }
+
+                    return { ...todo, completed: false };
+                  });
+
+                return newTodos;
+              });
+            }}
```

Comment:
create separate function
```suggestion
            onClick={onTodoAdd}
```